src/com/AbstractFactory/AbstractFactory.java

```diff
@@ -0,0 +1,27 @@
+package com.AbstractFactory;
+
+public class AbstractFactory {
+	
+	public SQLFactory getConnection(String acc) {
+		if(acc.equalsIgnoreCase("JDBC"))
+			return new JDBC();
+		if(acc.equalsIgnoreCase("OLEDB"))
```

Comment:
The equals comparison is better to do in the inverse way
`if (OLEDB.equalsIgnoreCase(acc))`
This is **Null Safety** because the parameter (acc) can be null